api/schema/deployment_resource.go

```diff
@@ -0,0 +1,67 @@
+package schema
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var MetadataType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"cloud": types.StringType,
+		"id":    types.StringType,
+	},
+}
+var TagType = types.ObjectType{
+	AttrTypes: map[string]attr.Type{
+		"key":   types.StringType,
+		"value": types.StringType,
+		"cloud": types.StringType,
+	},
+}
+
+var (
+	deploymentResource = Resource{
+		TypeName: "deployment",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Groups resources under Cloudsecure deployments",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"environment": resource_schema.StringAttribute{
+					MarkdownDescription: "Name of the deployment",
+					Required:            true,
+				},
+				"description": resource_schema.StringAttribute{
+					MarkdownDescription: "Description of Cloudsecure deployment",
+					Optional:            true,
+				},
+				"accounts": resource_schema.ListAttribute{
+					Optional:    true,
+					Description: "cloud accounts that need be a part of this deployment",
+					ElementType: MetadataType,
```

Comment:
This is inconsistent with the schemas of this provider's resources that manage accounts / subscriptions.
Split this attribute into two attributes:
* `aws_account_ids`, as a simple list of strings that contain AWS account IDs.
* `azure_subscription_ids`, as a simple list of strings that contains Azure subscription IDs.